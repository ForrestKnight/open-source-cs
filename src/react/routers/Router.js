/* Packages */
import React from 'react';
import {connect} from 'react-redux';
import {BrowserRouter, Route, Switch} from 'react-router-dom';


/* Components */
import Header from '../components/common/Header';
import Footer from '../components/common/Footer';


/* Pages/Routes */
import Default from '../components/pages/Default';
import Wildcard from '../components/pages/Wildcard';
import StyleGuide from '../components/pages/StyleGuide';


/* Router */
export class Router extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            loaded: false,
            loading: false
        };
    };


    componentWillMount() {
        const $pageLoader = document.querySelector('.page-loader');

        // Disable loader
        if ($pageLoader) {
            $pageLoader.setAttribute('data-loading', true);
            this.setState({
                loading: true
            });
        };

        // Fade in React app
        setTimeout(() => {
            $pageLoader.setAttribute('data-loaded', true);
            this.setState({
                loaded: true
            });
        }, 300);
    };


    render() {
        return (
            <BrowserRouter>
                <div 
                    id="Router"
                    data-page_ID={this.props.page_ID}
                    data-loading={this.state.loading}
                    data-loaded={this.state.loaded}
                >
                    <Header />

                    <Switch>
                        <Route 
                            path="/"
                            component={Default}
                            exact
                        />

                        <Route 
                            path="/StyleGuide"
                            component={StyleGuide}
                            exact
                        />

                        <Route 
                            path="/*"
                            component={Wildcard}
                        />
                    </Switch>

                    <Footer />
                </div>
            </BrowserRouter>
        );
    };
};


/* Connect to store */
const mapStateToProps = (state) => {
    return {
        page_ID: state.page.id
    };
};


export default connect(mapStateToProps, undefined)(Router);