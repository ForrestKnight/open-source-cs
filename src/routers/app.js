/* Packages */
const express = require('express');
const path = require('path');
const chalk = require('chalk');


/* Local files */
const publicPath = path.join(__dirname, '../../public/');
const appRoute = path.join(publicPath, 'index.html');
const wildcardRoute = path.join(publicPath, '404.html');


/* Router config */
const router = new express.Router();


/* Routes */
router.get('/', (req, res) => {
    try {
        return res.sendFile(appRoute);
    
    } catch(e) {
        res.send();
        console.log(
            chalk.red('An error occured: '),
            '\n',
            `${e}`
        );
    }
});

router.get('/*', (req, res) => {
    try {
        return res.sendFile(appRoute);

    } catch(e) {
        res.send();
        console.log(
            chalk.red('An error occured: '),
            '\n',
            `${e}`
        );
    }
});


module.exports = router;