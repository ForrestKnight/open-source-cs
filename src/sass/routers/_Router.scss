#Router {
    opacity: 0;

    &[data-loading=true] {
        &[data-loaded=true] {
            transition: opacity 300ms;
            opacity: 1;
        }
    }
}