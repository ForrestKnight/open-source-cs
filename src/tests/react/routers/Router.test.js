/* Packages */
import React from 'react';
import {shallow} from 'enzyme';


/* Component */
import {Router} from '../../../react/routers/Router';


/* Test config */
let component, load;
beforeEach(() => {
    load = jest.fn();
    component = shallow(
        <Router
            load={load}
        />
    );
});


/* Tests */
test('Render Router', () => {
    expect(component).toMatchSnapshot();
});


test('Router initial state', () => {
    expect(component.state('loaded')).toBe(false);
    expect(component.state('loading')).toBe(false);
});


test('Router load function', () => {
    setTimeout(() => {
        expect(load).toHaveBeenCalled();
    }, 200);

    setTimeout(() => {
        expect(component.state('loaded')).toBe(true);
        expect(component.state('loading')).toBe(true);    
    }, 500);
});