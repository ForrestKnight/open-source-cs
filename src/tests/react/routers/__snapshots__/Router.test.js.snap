// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Router 1`] = `
<BrowserRouter>
  <div
    data-loaded={false}
    data-loading={false}
    id="Router"
  >
    <Header />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/StyleGuide"
      />
      <Route
        component={[Function]}
        path="/*"
      />
    </Switch>
    <Footer />
  </div>
</BrowserRouter>
`;
